haystack/nodes/answer_generator/openai.py

```diff
@@ -81,26 +84,75 @@ def __init__(
-                         If not supplied, the default from OpenAPI docs is used:
+                         If not supplied, the default from OpenAI API docs is used:
                          [["What is human life expectancy in the United States?", "78 years."]]
         :param stop_words: Up to 4 sequences where the API stops generating further tokens. The returned text does
                            not contain the stop sequence.
-                           If you don't provide it, the default from OpenAPI docs is used: ["\n", "<|endoftext|>"]
+                           If you don't provide it, the default from OpenAI API docs is used: ["\n", "<|endoftext|>"]
+        :param instruction_prompt: A PromptTemplate used to explain to the model how to generate answers given a
+            supplied `context` and `query` at runtime. An `example_context` and a list of `examples` are used to provide
+            the model with examples to help steer the model towards the tone and answer format you would like.
+            If not supplied, the default instruction prompt is:
+            ```python
+                PromptTemplate(
+                    name="question-answering",
+                    prompt_text="Please answer the question according to the above context."
+                                "\n===\nContext: $examples_context\n===\n$examples\n\n"
+                                "===\nContext: $context\n===\n$query",
+                    prompt_params=["examples_context", "examples", "context", "query"],
+                )
+            ```
+            An explanation of how variables (for example '$context') are substituted into the `prompt_text` refer to
+            the PromptTemplate [documentation](https://docs.haystack.deepset.ai/docs/prompt_node#template-structure).
+        :param context_join_str: The separation string to use to join the input documents together to create the context
+            that will be used by the PromptTemplate.
         """
         super().__init__(progress_bar=progress_bar)
-        if not examples_context:
+        if (examples is None and examples_context is not None) or (examples is not None and examples_context is None):
+            logger.warning(
+                "If providing examples or a examples_context, we recommend providing both of them "
+                "so the examples correctly refer to the examples_context."
+            )
+        if examples_context is None:
             examples_context = "In 2017, U.S. life expectancy was 78.6 years."
-        if not examples:
+        if examples is None:
             examples = [["What is human life expectancy in the United States?", "78 years."]]
-        if not stop_words:
+        if stop_words is None:
             stop_words = ["\n", "<|endoftext|>"]
+        if instruction_prompt is None:
+            instruction_prompt = PromptTemplate(
+                name="question-answering",
```

Comment:
let's pick a name that is not conflicting with the predefined templates here: https://github.com/deepset-ai/haystack/blob/bcf3bfdf790f41d0ac252a1639ce345beb777060/haystack/nodes/prompt/prompt_node.py#L617
"question-answering-with-examples" might fit here